endorsement/dao/shared_drive.py

```diff
@@ -70,48 +104,311 @@ def get_drive_quota(a):
     """
     return a shared drive quota model from OrgUnit
     """
-    drive_quota, _ = SharedDriveQuota.objects.get_or_create(org_unit=a.OrgUnit)
+    defaults = {"org_unit_name": a.org_unit_name}
+    try:
+        defaults["quota_limit"] = a.quota_limit
+    except ValueError:
+        "Drive has no quota set. Probably in invalid org unit."
+
+    drive_quota, _ = SharedDriveQuota.objects.update_or_create(
+        org_unit_id=a.org_unit_id,
+        defaults=defaults,
+    )
     return drive_quota
 
 
-def get_shared_drive_member(a):
+def get_shared_drive_member(a: GoogleDriveState):
     """
     return a shared drive member model from Member and Role
     """
     member = get_member(a)
     role = get_role(a)
     shared_drive_member, _ = SharedDriveMember.objects.get_or_create(
-        member=member, role=role)
+        member=member, role=role
+    )
     return shared_drive_member
 
 
-def get_member(a):
+def get_member(a: GoogleDriveState):
     """
     return a member model, bare netid or non-uw email
     """
-    netid = netid_regex.match(a.Member)
-    member, _ = Member.objects.get_or_create(
-        name=netid.group('netid') if netid else a.Member)
-    return member
+    netid = netid_regex.match(a.member)
+    name = netid.group("netid") if netid else a.member
 
+    return Member.objects.get_member(name=name)
 
-def get_role(a):
+
+def get_role(a: GoogleDriveState):
     """
-    return a role model
+    Return a Role model.
+
+    Raises SharedDriveNonPrivilegedMember IFF role is not a manager role.
     """
     # cull non-manager roles until others are interesting
-    if a.Role != SharedDriveRecord.MANAGER_ROLE:
+    if a.role != Role.MANAGER_ROLE:
+        msg = "Shared drive member {} is not a manager ({!r}) and instead {!r}"
         raise SharedDriveNonPrivilegedMember(
-            f"Shared drive member {a.Member} is not a manager")
+            msg.format(a.member, Role.MANAGER_ROLE, a.role),
+        )
 
-    role, _ = Role.objects.get_or_create(role=a.Role)
+    role, _ = Role.objects.get_or_create(role=a.role)
     return role
 
 
-class DataFromRow(object):
-    def __init__(self, *args, **kwargs):
-        columns = kwargs.get('columns', [])
-        row = kwargs.get('row', [])
+def get_subscription(sdr: SharedDriveRecord):
```

Comment:
@mikeseibel this method prompted the PR as I think it's getting dangerously close to code you're writing and I'd really rather only write it once 😆